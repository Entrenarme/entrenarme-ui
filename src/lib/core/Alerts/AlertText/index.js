// @flow
import * as React from 'react';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/pro-light-svg-icons';

import { regularFont } from '../../../helpers/fonts';
import colors from '../../../helpers/colors';

const setColor = (status, elementType) => {
  const colorType = elementType === 'icon' ? 'default' : 'light';

  switch (status) {
    case 'primary':
      return colors.primary[colorType];
    case 'secondary':
      return colors.secondary[colorType];
    case 'success':
      return colors.success[colorType];
    case 'warning':
      return colors.warning[colorType];
    default:
      return colors.error[colorType];
  }
};

const MainContainer = styled.div`
  font-family: ${regularFont};
  display: flex;
  animation: ${props => props.options.hideState && `fadeOut 1s forwards`};
  align-items: center;
  justify-content: space-between;
  border-radius: 4px;
  padding: 15px 20px;

  background-color: ${props => setColor(props.options.type)};

  > div > div {
    color: ${props => setColor(props.options.type, 'icon')};
  }

  @keyframes fadeOut {
    0% {
      opacity: 1;
    }

    30% {
      opacity: 0;
    }

    100% {
      height: 0px;
      opacity: 0;
    }
  }

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (max-width: 767px) {
    > div > div {
      align-self: flex-start;
    }

    padding: 15px 20px;
    align-items: flex-start;
  }
`;

const RequiredContent = styled.div`
  display: flex;
  align-items: center;
`;

const IconContainer = styled.div`
  font-size: 1.375rem;
  margin-right: 20px;
  align-self: flex-start;

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (max-width: 767px) {
    font-size: 1.125rem;
    margin-right: 14px;
  }
`;

const TextContainer = styled.div`
  font-size: 1rem;
  color: ${colors.text};

  a {
    color: ${colors.secondary.default} !important;
    text-decoration: none !important;
  }

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (max-width: 767px) {
    font-size: 0.875rem;
  }
`;

const CloseIconContainer = styled.div`
  font-size: 1.375rem;
  cursor: pointer;
  margin-left: 20px;
  color: ${colors.text};
  align-self: flex-start;

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (max-width: 767px) {
    font-size: 1.125rem;
  }
`;

type Props = {
  icon: React.Node,
  text: React.Node,
  link?: React.Node,
  type?: 'primary' | 'secondary' | 'success' | 'warning' | 'error',
  canClose?: boolean,
  onClose?: Function,
};

type State = {
  hideState: boolean,
};

class AlertText extends React.Component<Props, State> {
  state = {
    hideState: false,
  };

  render() {
    const { icon, text, link, type, canClose, ...rest } = this.props;
    const { hideState } = this.state;

    return (
      <MainContainer options={{ type, hideState }} {...rest}>
        <RequiredContent>
          <IconContainer>{icon}</IconContainer>
          <TextContainer>
            {text} {link}
          </TextContainer>
        </RequiredContent>
        {canClose ? (
          <CloseIconContainer
            onClick={() => {
              this.setState({ hideState: true });
            }}
          >
            <FontAwesomeIcon icon={faTimes} />
          </CloseIconContainer>
        ) : null}
      </MainContainer>
    );
  }
}

AlertText.defaultProps = {
  type: 'primary',
  canClose: false,
  onClose: () => {},
};

export default AlertText;
