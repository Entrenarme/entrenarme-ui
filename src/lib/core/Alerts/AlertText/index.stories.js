// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAddressBook } from '@fortawesome/pro-light-svg-icons';

import AlertText from './index';

const actions = {
  onClick: action('onClick'),
};

const content =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin aliquet nulla a consectetur porta. In massa lacus, condimentum a turpis at, rutrum suscipit metus. Morbi sit amet odio eget tortor bibendum pulvinar. Proin varius ac leo eu bibendum. Etiam eu lectus non nibh gravida mollis.';

storiesOf('Alerts/Alert text', module).add(
  'default',
  withInfo('default')(() => (
    <AlertText text={content} icon={<FontAwesomeIcon icon={faAddressBook} />} />
  )),
);

storiesOf('Alerts/Alert text', module).add(
  'with canClose',
  withInfo('with canClose')(() => (
    <AlertText
      {...actions}
      text={content}
      icon={<FontAwesomeIcon icon={faAddressBook} />}
      canClose
    />
  )),
);

storiesOf('Alerts/Alert text', module).add(
  'with link and canClose',
  withInfo('with link and canClose')(() => (
    <AlertText
      {...actions}
      text={content}
      icon={<FontAwesomeIcon icon={faAddressBook} />}
      link={
        <a
          href="https://www.google.com"
          target="_blank"
          rel="noopener noreferrer"
        >
          Mas info
        </a>
      }
      canClose
    />
  )),
);

storiesOf('Alerts/Alert text', module).add(
  'warning type',
  withInfo('warning type')(() => (
    <AlertText
      {...actions}
      text={content}
      icon={<FontAwesomeIcon icon={faAddressBook} />}
      type="warning"
      canClose
    />
  )),
);

storiesOf('Alerts/Alert text', module).add(
  'danger type',
  withInfo('danger type')(() => (
    <AlertText
      {...actions}
      text={content}
      icon={<FontAwesomeIcon icon={faAddressBook} />}
      type="error"
      canClose
    />
  )),
);
