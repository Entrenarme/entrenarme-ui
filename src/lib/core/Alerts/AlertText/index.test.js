import * as React from 'react';
import { render, getByText } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAlarmClock } from '@fortawesome/free-solid-svg-icons';

import AlertText from './index';

const text = 'Test text';

const { container } = render(
  <AlertText text={text} icon={<FontAwesomeIcon icon={faAlarmClock} />} />,
);

test('renders "AlertText"', () => {
  expect(container.firstChild).toMatchSnapshot();
});
