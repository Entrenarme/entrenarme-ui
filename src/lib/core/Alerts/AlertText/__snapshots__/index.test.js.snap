// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "AlertText" 1`] = `
<div
  class="sc-bdVaJa ieWZKV"
>
  <div
    class="sc-bwzfXH fNOhzS"
  >
    <div
      class="sc-htpNat lphCPo"
    />
    <div
      class="sc-bxivhb iyyakI"
    >
      Test text
    </div>
  </div>
</div>
`;
